src/lib/runtime/Web3FunctionRunner.ts

```diff
@@ -287,9 +289,35 @@ export class Web3FunctionRunner {
     const mountPath = randomUUID();
     const serverPort =
       options.serverPort ?? (await Web3FunctionNetHelper.getAvailablePort());
+
+    const httpProxyPort = await Web3FunctionNetHelper.getAvailablePort();
+    const httpProxyHost =
+      options.runtime === "thread" ? "127.0.0.1" : "host.docker.internal";
+    this._httpProxy = new Web3FunctionHttpProxy(
+      options.downloadLimit,
+      options.uploadLimit,
+      (url: URL) => {
+        if (options.blacklistedHosts) {
+          return options.blacklistedHosts.includes(url.hostname);
+        }
+
+        return false;
+      },
```

Comment:
I think it could be reduced with something like that
```suggestion
      (url: URL) => options.blacklistedHosts?.includes(url.hostname),
```

Reply:
updated in 9bd80a257f1a6589665050337c9a05943b6eea6e